Sources/GRPC/Compression/Zlib.swift

```diff
@@ -0,0 +1,422 @@
+/*
+ * Copyright 2020, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import CGRPCZlib
+import NIO
+import struct Foundation.Data
+
+enum Zlib {
+  // MARK: Deflate (compression)
+
+  /// Creates a new compressor for the given compression format.
+  ///
+  /// This compressor is only suitable for compressing whole messages at a time. Callers
+  /// must `reset()` the compressor between subsequent calls to `deflate`.
+  ///
+  /// - Parameter format:The expected compression type.
+  class Deflate {
+    private var stream: ZStream
+    private let format: CompressionFormat
+
+    init(format: CompressionFormat) {
+      self.stream = ZStream()
+      self.format = format
+      self.initialize()
+    }
+
+    deinit {
+      self.end()
+    }
+
+    /// Decompresses the data in `input` into the `output` buffer.
```

Comment:
Wait, shouldn't this method be about *compressing*?

Reply:
Good catch, thanks!